### PR TITLE
Fully toml user input

### PR DESCRIPTION
**Pull request by teddygroves**
This PR changes the user input to Maud so that it consists entirely of `.toml` files. Whereas custom inits, measurements, independent priors and multivariate priors all used to have their own custom csv formats, after this change they are all in toml.

There are a few reasons why I thought this would be a good idea:

- Fewer files (experiments and measurements can go in the same file, as can independent and multivariate priors)
- Fewer custom formats for the user to learn (e.g. inits and priors have basically the same format)
- Neater correspondence with Maud's data model makes validation easier and allows a good amount (~3000 lines) of code deletion.
- toml format makes it less clunky to add optional fields to priors. This inches us a bit closer to the day when we can specify arbitrary known parameter values in the priors file.
- I think toml files are easier to programmatically generate than csvs.

I think some further consolidation is probably desirable, for example changing the priors file to a parameters file that can also include inits. I thought it was better to go one step at a time though.

A few issues came up while changing the code, which I think I fixed:

- a weird bug when loading inferencedatas using `from_json`: the experiment and enzyme dimensions of the `conc_enzyme` variable are swapping round on the current master for some reason that I couldn't reproduce. It seems to be ok in this version but I don't know why.
- The safety checker has found a vulnerability in a dependency that hasn't yet been patched: [https://pyup.io/v/51549/f17](https://pyup.io/v/51549/f17). I think it is safe to ignore so I added this to the ignored list in `tox.ini`.
- I updated or deleted a lot of stale docs and data files.
- I shut up a pandas warning by explicitly specifying some dtypes
- I removed `StanData` from Maud's data model as it was a big pain and I didn't really see what it was adding.
- A lot of other non-user-facing data model changes generally aiming at less code and more work done by pydantic.

Checklist:

- [x] Updated any relevant documentation
- [ ] Add an adr doc if appropriate
- [ ] Include links to any relevant issues in the description
- [x] Unit tests passing
- [x] Integration tests passing

### CODE REVIEW

**Review by carrascomj — COMMENTED** [review]
I left a comment about pydantic but otherwise looks good. I need to test my model with the new configuration.

File: src/maud/data_model/prior.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        """Check that covariance matrix is positive definite."""
+        try:
+            np.linalg.cholesky(v)
+        except LinAlgError as e:
+            raise ValueError(
+                "Covariance matrix is not positive definite"
+            ) from e
+        return v
+
+    @root_validator
```
[the remark this answers is not in the document]
Comment (teddygroves) [reply src/maud/data_model/prior.py:143]: Thanks, I had that problem a few times and was wondering how to fix it.

File: src/maud/getting_priors.py [hunk 2]
```diff
@@ -1,135 +1,136 @@
...
+
+def get_ind_prior_2d(
+    pi: Optional[List[IndPriorAtomInput]],
+    ids: List[List[str]],
+    id_components: List[List[IdComponent]],
+    non_negative: bool,
+    default_loc: float,
+    default_scale: float,
+) -> IndPrior2d:
+    """Get an independent 1d prior from a prior input and StanVariable."""
```
Comment (carrascomj) [inline src/maud/getting_priors.py:93]:
```suggestion
    """Get an independent 2d prior from a prior input and StanVariable."""
```

File: tests/data/drain/experiments.toml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+[[experiment]]
+id = "condition1"
+is_train = true
+is_test = true
+measurements = [
+  {target_type = "mic", metabolite = "M1", compartment = "c", value = 1.44218, error_scale = 0.1},
+  {target_type = "mic", metabolite = "M2", compartment = "c", value = 1.43082, error_scale = 0.1},
+  {target_type = "mic", metabolite = "M2", compartment = "e", value = 1, error_scale = 0.05},
+  {target_type = "flux", reaction = r3, value = 0.125573, error_scale = 0.01},
+]p
```
Comment (carrascomj) [inline tests/data/drain/experiments.toml:10]:
```suggestion
]
```
How is that not failing parsing?

File: tests/data/linear/experiments.toml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+[[experiment]]
+id = "condition1"
+is_train = true
+is_test = true
+temperature = 299.0
+measurements = [
+  {target_type = "mic", metabolite = "M1", compartment = "c", value = 0.59, error_scale = 0.1},
+  {target_type = "mic", metabolite = "M2", compartment = "c", value = 0.38, error_scale = 0.1},
+  {target_type = "mic", metabolite = "M1", compartment = "e", value = 1.09, error_scale = 0.05}.
```
Comment (carrascomj) [inline tests/data/linear/experiments.toml:9]:
```suggestion
  {target_type = "mic", metabolite = "M1", compartment = "e", value = 1.09, error_scale = 0.05},
```

**Review by carrascomj — APPROVED** [review]
LGTM, just pointed out some typos.

**Pull request merged**